### PR TITLE
[Draft] Download lib with the init

### PR DESCRIPTION
**Pull request by Miniplop**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/init/src/index.ts [hunk 1]
```diff
@@ -49,8 +52,14 @@ async function init() {
   // install the required dependencies in the project.
   await installRequiredDependencies(cwd, frameworkResult.value);
 
+  const sliceLibPath = await (() => {
```
Comment (Miniplop) [inline packages/init/src/index.ts]: We can change this notation by 
 
`
const sliceLibPath = lib ? await installLib(cwd, lib, branch) : null
`

The default case is explicit and we avoid the IIFE overload

File: packages/init/src/steps/install-lib.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    const pkgJson = PackageJsonHelper.fromPath(
+      path.join(projectPath, "package.json")
+    );
+    if (pkgJson instanceof Error) throw pkgJson;
+
+    const dependencies = Dependencies.fromPkgFormat(pkgJson.dependencies);
+    if (dependencies) await pkgManager.install(dependencies);
+
+    // generate meta file
+    Files.write(path.join(libDestinationFolder, "meta.json"), {
```
Comment (Miniplop) [inline packages/init/src/steps/install-lib.ts:79]: what is this file for ? What happen if the user doesn't have this file or delete it ?

File: packages/init/src/steps/install-lib.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      ({ path }) => {
+        return `~/${name}/${path.replace("src/", "")}`;
+      }
+    );
+
+    spinner.succeed(
+      `Slice library "${libGithubPath}" was installed successfully`
+    );
+
+    // If the libs have multiple slice libs here we are grouping them into one
```
Comment (Miniplop) [inline packages/init/src/steps/install-lib.ts]: not true anymore

File: packages/init/src/steps/install-lib.ts (hunk 2 above)
Comment (arnaudlewis) [reply packages/init/src/steps/install-lib.ts:79]: Used to identify the lib that was downloaded so we can track it on amplitude later on, on SM

File: packages/init/src/steps/install-lib.ts (hunk 3 above)
Comment (arnaudlewis) [reply packages/init/src/steps/install-lib.ts]: true

File: packages/init/src/index.ts (hunk 1 above)
Comment (arnaudlewis) [reply packages/init/src/index.ts]: it's undefined not null here but yeah if you prefer

**Pull request merged**